### PR TITLE
klient: add set operations on machines

### PR DESCRIPTION
**Pull request by ppknap**
This PR adds package with basic operations on machine sets.

## Motivation and Context
Needed for `machine mount` subcommand. 

## How Has This Been Tested?
Unit tests.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: go/src/koding/klient/machine/machinegroup/idset/idset_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+	"koding/klient/machine"
+)
+
+func TestUnion(t *testing.T) {
+	tests := map[string]struct {
+		A, B, Expected []machine.ID
+	}{
+		"simple union": {
+			A:        genIDs("x1, x2, x3, x4"),
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/idset/idset_test.go]: What's wrong with the above?

```
A: []machine.ID{"x1", "x2", "x3", "x4"},
```

Comment (ppknap) [inline go/src/koding/klient/machine/machinegroup/idset/idset_test.go]: I found that approach more readable and it was easier to modify when I replaced `[]machine.ID` with `machine.IDSlice` (see #10054). It's a matter of taste, If you want I can change it.

Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/idset/idset_test.go]: I'd prefer initialising slices directly here.

**Review by rjeczalik — APPROVED** [review]

File: go/src/koding/klient/machine/machinegroup/idset/idset.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	return res
+}
+
+// Intersection returns a set that contains all elements of A that also belong
+// to B, but no other elements.
+func Intersection(a, b []machine.ID) []machine.ID {
+	aset := make(map[machine.ID]bool)
+	for i := range a {
+		aset[a[i]] = true
```
Comment (cihangir) [inline go/src/koding/klient/machine/machinegroup/idset/idset.go]: you could use struct{} here too, to be consistent with other methods.

**Review by cihangir — COMMENTED** [review]
lgtm

File: go/src/koding/klient/machine/machinegroup/idset/idset.go (hunk 2 above)
Comment (cihangir) [inline go/src/koding/klient/machine/machinegroup/idset/idset.go]: no it is not strictly needed

Comment (ppknap) [inline go/src/koding/klient/machine/machinegroup/idset/idset.go]: ah, you want me to assign `struct{}{}` and then check `_, ok := ...; ok`. will add this change in one of next PRs

**Pull request merged**